### DISCUSSION TITLE
Duplicating pipelines

### DISCUSSION THREAD

**Discussion by EmilyGraceSeville7cf** [community/community#26825] [Actions]
<p>Hello! I have <a href="https://github.com/alvinseville7cf/TestCI/blob/854ab6205adb9d982783a02d588be8e5d0abe15c/.github/workflows/ci.yml" rel="noopener nofollow ugc">.yml</a> config with the following first lines:</p>
<pre><code class="lang-auto">name: CI

on:
  push:
    branches: [ main ]
    tags: 'v*'
  pull_request:
    branches: [ main ]

  workflow_dispatch:
</code></pre>
<p>When I push tag I obtain duplicating pipelines (please checkout the last workflow run):<br>
<img src="https://user-images.githubusercontent.com/42812113/181100061-8d679522-be69-4fc9-9e9b-6a19252bf5f7.png" alt="image" data-base62-sha1="7KcHeU6nrucnmpqEbmFvNTGmB8a" width="411" height="272"><br>
I have <code>git config --global push.followTags true</code> enabled.</p>
<p>How to fix it? My test repo is <a href="https://github.com/alvinseville7cf/TestCI" rel="noopener nofollow ugc">this</a> one. I want:</p>
<ul>
<li>automatically check main.sh on every commit/tag push to <code>main</code> branch</li>
<li>automatically generate installer on every v* tag push to <code>main</code> branch</li>
</ul>

**Comment by BelKed** [comment]
<p>I have same problem: <a href="https://github.community/t/how-to-trigger-an-action-on-push-or-create-tag-but-not-on-both/161563" class="inline-onebox">How to trigger an action on push or create tag but not on both?</a></p>

**Comment by EmilyGraceSeville7cf** [comment, marked as answer]
<aside class="onebox githubfolder" data-onebox-src="https://github.com/emilyseville7cfg/CSharp---Exercise---Other---Terminal-reflection-client/tree/v0.1/.github/workflows">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/emilyseville7cfg/CSharp---Exercise---Other---Terminal-reflection-client/tree/v0.1/.github/workflows" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>

  <article class="onebox-body">
    <h3><a href="https://github.com/emilyseville7cfg/CSharp---Exercise---Other---Terminal-reflection-client/tree/v0.1/.github/workflows" target="_blank" rel="noopener nofollow ugc">CSharp---Exercise---Other---Terminal-reflection-client/.github/workflows at v0.1...</a></h3>

  <p><a href="https://github.com/emilyseville7cfg/CSharp---Exercise---Other---Terminal-reflection-client/tree/v0.1/.github/workflows" target="_blank" rel="noopener nofollow ugc">v0.1/.github/workflows</a></p>

  <p><span class="label1">Terminal reflection client. Contribute to emilyseville7cfg/CSharp---Exercise---Other---Terminal-reflection-client development by creating an account on GitHub.</span></p>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>Here I’ve solved this problem by creating two different .yml configs. I don’t think it’s perfect solution but it is solution. :slight_smile:</p>
